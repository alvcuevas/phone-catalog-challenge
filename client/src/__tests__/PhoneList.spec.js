import { render } from "@testing-library/react";
import PhoneListContainer from "../components/phone-list";

it("should renders an array of phones", () => {
  const mockedPhones = [
    {
      id: 1,
      phone: "Galaxy Note 20",
      desc: "Samsung Galaxy Note 20 Rose 256GB",
      network: "GSM / CDMA / HSPA / EVDO / LTE",
      dimensions: "150.9 x 75.7 x 8.3 mm (5.94 x 2.98 x 0.33 in)",
      weight: "194 g (6.84 oz)",
      type: "IPS LCD capacitive touchscreen, 16M colors",
      size: "6.1 inches, 90.3 cm2 (~79.0% screen-to-body ratio)",
      resolution: "828 x 1792 pixels, 19.5:9 ratio (~326 ppi density)",
      colors: "Rose Gold",
      price: "877,37€",
      img:
        "https://www.maxmovil.com/media/catalog/product/cache/1/small_image/200x200/9df78eab33525d08d6e5fb8d27136e95/g/a/galaxy-note-20-4g_0002_81s7yrb1zll._ac_sl1500_.jpg",
    },
    {
      id: 2,
      phone: "Xiaomi Redmi 9C",
      desc: "Xiaomi Redmi 9C Black 3GB/64GB",
      network: "GSM / HSPA / LTE",
      dimensions: "138.4 x 67.3 x 7.3 mm (5.45 x 2.65 x 0.29 in)",
      weight: "148 g (5.22 oz)",
      type: "LED-backlit IPS LCD, capacitive touchscreen, 16M colors",
      size: "4.7 inches, 60.9 cm2 (~65.4% screen-to-body ratio)",
      resolution: "750 x 1334 pixels, 16:9 ratio (~326 ppi density)",
      colors: "Black",
      price: "148€",
      img:
        "https://www.maxmovil.com/media/catalog/product/cache/1/small_image/200x200/9df78eab33525d08d6e5fb8d27136e95/r/e/redmi-9c-gris_0002_61o-icdwxal._ac_sl1000__1.jpg",
    },
  ];

  const renderedList = render(<PhoneListContainer list={mockedPhones} />);
  expect(renderedList).toMatchSnapshot();
});
