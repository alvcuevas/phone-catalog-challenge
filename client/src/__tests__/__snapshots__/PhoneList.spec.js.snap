// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders an array of phones 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-container-1"
      >
        <div
          class="makeStyles-catalog-2"
        >
          <div
            class="MuiPaper-root MuiCard-root makeStyles-root-3 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <button
              class="MuiButtonBase-root MuiCardActionArea-root"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiCardMedia-root makeStyles-media-4"
                style="background-image: url(https://www.maxmovil.com/media/catalog/product/cache/1/small_image/200x200/9df78eab33525d08d6e5fb8d27136e95/g/a/galaxy-note-20-4g_0002_81s7yrb1zll._ac_sl1500_.jpg);"
                title="Galaxy Note 20"
              />
              <div
                class="MuiCardContent-root makeStyles-content-6"
              >
                <h2
                  class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M9 11.3l3.71 2.7-1.42-4.36L15 7h-4.55L9 2.5 7.55 7H3l3.71 2.64L5.29 14z"
                      transform="scale(1.33, 1.33)"
                    />
                  </svg>
                   
                  Galaxy Note 20
                </h2>
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
                >
                  Samsung Galaxy Note 20 Rose 256GB
                </p>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <div
              class="MuiCardActions-root makeStyles-actions-7 MuiCardActions-spacing"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-5 MuiButton-outlinedPrimary MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <span
                    class="MuiButton-startIcon MuiButton-iconSizeSmall"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                      />
                    </svg>
                  </span>
                   
                  Details
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                aria-label="share"
                class="MuiButtonBase-root MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiPaper-root MuiCard-root makeStyles-root-3 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <button
              class="MuiButtonBase-root MuiCardActionArea-root"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiCardMedia-root makeStyles-media-4"
                style="background-image: url(https://www.maxmovil.com/media/catalog/product/cache/1/small_image/200x200/9df78eab33525d08d6e5fb8d27136e95/r/e/redmi-9c-gris_0002_61o-icdwxal._ac_sl1000__1.jpg);"
                title="Xiaomi Redmi 9C"
              />
              <div
                class="MuiCardContent-root makeStyles-content-6"
              >
                <h2
                  class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M9 11.3l3.71 2.7-1.42-4.36L15 7h-4.55L9 2.5 7.55 7H3l3.71 2.64L5.29 14z"
                      transform="scale(1.33, 1.33)"
                    />
                  </svg>
                   
                  Xiaomi Redmi 9C
                </h2>
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
                >
                  Xiaomi Redmi 9C Black 3GB/64GB
                </p>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <div
              class="MuiCardActions-root makeStyles-actions-7 MuiCardActions-spacing"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-5 MuiButton-outlinedPrimary MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <span
                    class="MuiButton-startIcon MuiButton-iconSizeSmall"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                      />
                    </svg>
                  </span>
                   
                  Details
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                aria-label="share"
                class="MuiButtonBase-root MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-container-1"
    >
      <div
        class="makeStyles-catalog-2"
      >
        <div
          class="MuiPaper-root MuiCard-root makeStyles-root-3 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <button
            class="MuiButtonBase-root MuiCardActionArea-root"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiCardMedia-root makeStyles-media-4"
              style="background-image: url(https://www.maxmovil.com/media/catalog/product/cache/1/small_image/200x200/9df78eab33525d08d6e5fb8d27136e95/g/a/galaxy-note-20-4g_0002_81s7yrb1zll._ac_sl1500_.jpg);"
              title="Galaxy Note 20"
            />
            <div
              class="MuiCardContent-root makeStyles-content-6"
            >
              <h2
                class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9 11.3l3.71 2.7-1.42-4.36L15 7h-4.55L9 2.5 7.55 7H3l3.71 2.64L5.29 14z"
                    transform="scale(1.33, 1.33)"
                  />
                </svg>
                 
                Galaxy Note 20
              </h2>
              <p
                class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
              >
                Samsung Galaxy Note 20 Rose 256GB
              </p>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight"
            />
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiCardActions-root makeStyles-actions-7 MuiCardActions-spacing"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-5 MuiButton-outlinedPrimary MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <span
                  class="MuiButton-startIcon MuiButton-iconSizeSmall"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                    />
                  </svg>
                </span>
                 
                Details
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="share"
              class="MuiButtonBase-root MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiCard-root makeStyles-root-3 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <button
            class="MuiButtonBase-root MuiCardActionArea-root"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiCardMedia-root makeStyles-media-4"
              style="background-image: url(https://www.maxmovil.com/media/catalog/product/cache/1/small_image/200x200/9df78eab33525d08d6e5fb8d27136e95/r/e/redmi-9c-gris_0002_61o-icdwxal._ac_sl1000__1.jpg);"
              title="Xiaomi Redmi 9C"
            />
            <div
              class="MuiCardContent-root makeStyles-content-6"
            >
              <h2
                class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9 11.3l3.71 2.7-1.42-4.36L15 7h-4.55L9 2.5 7.55 7H3l3.71 2.64L5.29 14z"
                    transform="scale(1.33, 1.33)"
                  />
                </svg>
                 
                Xiaomi Redmi 9C
              </h2>
              <p
                class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
              >
                Xiaomi Redmi 9C Black 3GB/64GB
              </p>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight"
            />
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiCardActions-root makeStyles-actions-7 MuiCardActions-spacing"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-5 MuiButton-outlinedPrimary MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <span
                  class="MuiButton-startIcon MuiButton-iconSizeSmall"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                    />
                  </svg>
                </span>
                 
                Details
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="share"
              class="MuiButtonBase-root MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
